gpii/node_modules/spiSettingsHandler/src/SpiSettingsHandler.js

```diff
@@ -413,11 +413,35 @@ gpii.windows.spi.setImpl = function (payload) {
     // to reload the default theme (or last saved?) - causing any custom wallpaper to be lost.
     // This hack just gets the current high-contrast state and does nothing if it's already turned off.
     var applyPromise;
-    if (payload.settings.HighContrastOn && !payload.settings.HighContrastOn.value &&
-        !payload.settings.HighContrastTheme) {
-        var currentState = gpii.windows.spi.getImpl(fluid.copy(payload));
-        if (!currentState.HighContrastOn.value) {
+    if (payload.settings.HighContrastOn) {
+        var doNothing = false;
+        if (!payload.settings.HighContrastOn.value && !payload.settings.HighContrastTheme) {
+            var currentState = gpii.windows.spi.getImpl(fluid.copy(payload));
+            if (!currentState.HighContrastOn.value) {
+                doNothing = true;
+            }
+        }
+
+        if (doNothing) {
             applyPromise = fluid.promise().resolve();
+        } else {
+            var cachedFilesPathEnv = "%APPDATA%\\Microsoft\\Windows\\Themes\\CachedFiles";
```

Comment:
`cachedFilePath = path.join(env.APPDATA, "Mic..")` is sufficient.

(same goes for the transcoded path below)

Reply:
Doing this